build OpenBLAS for Android-x86_64 failed

https://github.com/xianyi/OpenBLAS/archive/v0.3.18.tar.gz

android-ndk 23.0.7599858

error log：
```bash
make -w -j6 OSNAME=Android ARCH=x86_64 TARGET=ATOM HOSTCC=/data/data/com.termux/files/usr/bin/cc ONLY_CBLAS=1 ARM_SOFTFP_ABI=1 CC=x86_64-linux-android21-clang AS=x86_64-linux-android-as AR=llvm-ar
make: Entering directory '/data/data/com.termux/files/usr/tmp/tmp.sZAqS1WN2a' 
clang-13: error: the clang compiler does not support '-march=native'
clang-13: error: the clang compiler does not support '-march=native'
clang-13: error: the clang compiler does not support '-march=native'    
make: *** [Makefile.prebuild:70: getarch] Error 1                       
Makefile.system:267: Makefile.conf: No such file or directory
make: *** No rule to make target 'Makefile.conf'.  Stop.
```

Makefile.system L188-L193
```bash
# On x86_64 build getarch with march=native unless the compiler is PGI. This is required to detect AVX512 support in getarch.
ifeq ($(HOSTARCH), x86_64)
ifeq ($(findstring pgcc,$(HOSTCC))$(findstring nvc,$(HOSTCC)),)
GETARCH_FLAGS += -march=native
endif
endif
```

--- a/Makefile.system
+++ b/Makefile.system
@@ -9,11 +9,10 @@ ifndef TOPDIR
 TOPDIR = . 
 endif
 
- # If ARCH is not set, we use the host system's architecture for getarch compile options.
-ifndef ARCH
+# we need to use the host system's architecture for getarch compile options even especially when cross-compiling
 HOSTARCH := $(shell uname -m)
-else
-HOSTARCH = $(ARCH)
+ifeq ($(HOSTARCH), amd64)
+HOSTARCH=x86_64
 endif
 
 HAVE_GAS := $(shell as -v < /dev/null 2>&1 | grep GNU 2>&1 >/dev/null)

